content/fundamentals/module-reference.md

```diff
@@ -191,17 +191,32 @@ const catsRepository = await this.moduleRef.resolve(CatsRepository, contextId);
 
 #### Instantiating classes dynamically
 
-To dynamically instantiate a class that wasn't previously registered as a provider, use the module reference's `create()` method.
+To dynamically instantiate a class that wasn't previously registered as a provider but depends on other providers use the module reference's `create()` method.
+
+```typescript
+@@filename(cat)
+@Injectable({scope: Scope.TRANSIENT})
+export class Cat {
+  private name: string;
+
+  constructor(private logger: Logger) {}
+  
+  init(name: string) {
+    this.name = name;
+  }
+}
+```
 
 ```typescript
 @@filename(cats.service)
 @Injectable()
 export class CatsService implements OnModuleInit {
-  private catsFactory: CatsFactory;
+  private myCat: Cat;
   constructor(private moduleRef: ModuleRef) {}
 
   async onModuleInit() {
-    this.catsFactory = await this.moduleRef.create(CatsFactory);
+    this.myCat = await this.moduleRef.create(Cat);
+    this.myCat.init('Meowth');
```

Comment:
Can we revert this change? With the text update you proposed, everything should be clear now

Reply:
Creating a factory was kind of confusing to me. I think it should be clear that when you can create a class, you can also create it using a factory. Or am I wrong?

Reply:
`Factory` acts as a random (sample) name here. We can change it to, let's say, `CatsHelpers` or `CatsStorage`, or `CatsUtils`. I refer to the rest of the additional code (which seems unnecessary to me)